{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Purpose of script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Input data:** \n",
    "- KNN Imputer filled full TCGA dataset - contains only the cancer type and protein expression\n",
    "\n",
    "**Output data:** \n",
    "- Machine learning (70%) and validation (30%) TCGA datasets with shortlisted protein list - contains the cancer type and validated proteins as defined in the following paper: Akbani, R., Ng, P., Werner, H. et al. A pan-cancer proteomic perspective on The Cancer Genome Atlas. Nat Commun 5, 3887 (2014). https://doi.org/10.1038/ncomms4887\n",
    "- Unique gene list - contains all the gene which are associated with the shortlisted proteins, this will be used to create the distance matrix\n",
    "- Protein to gene conversion list - contains the shortlisted protein list and the gene or genes that the protein is associated with, this will be used to creat the distance matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Importing data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Importing necessary packages\n",
    "import pandas as pd \n",
    "import numpy as np "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project_id</th>\n",
       "      <th>1433BETA</th>\n",
       "      <th>1433EPSILON</th>\n",
       "      <th>1433ZETA</th>\n",
       "      <th>4EBP1</th>\n",
       "      <th>4EBP1_pS65</th>\n",
       "      <th>4EBP1_pT37T46</th>\n",
       "      <th>4EBP1_pT70</th>\n",
       "      <th>53BP1</th>\n",
       "      <th>ACC_pS79</th>\n",
       "      <th>...</th>\n",
       "      <th>XPF</th>\n",
       "      <th>XRCC1</th>\n",
       "      <th>YAP</th>\n",
       "      <th>YAP_pS127</th>\n",
       "      <th>YB1</th>\n",
       "      <th>YB1_pS102</th>\n",
       "      <th>YTHDF2</th>\n",
       "      <th>YTHDF3</th>\n",
       "      <th>ZAP.70</th>\n",
       "      <th>ZEB1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5424</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.066729</td>\n",
       "      <td>0.412330</td>\n",
       "      <td>-0.229260</td>\n",
       "      <td>-0.034305</td>\n",
       "      <td>-0.030930</td>\n",
       "      <td>0.71239</td>\n",
       "      <td>-0.240950</td>\n",
       "      <td>-0.916615</td>\n",
       "      <td>-0.285390</td>\n",
       "      <td>...</td>\n",
       "      <td>0.436293</td>\n",
       "      <td>-0.361380</td>\n",
       "      <td>0.106262</td>\n",
       "      <td>-0.529325</td>\n",
       "      <td>-0.530005</td>\n",
       "      <td>0.002020</td>\n",
       "      <td>-0.397034</td>\n",
       "      <td>-0.382803</td>\n",
       "      <td>-1.629115</td>\n",
       "      <td>1.311241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5207</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.218150</td>\n",
       "      <td>-0.096406</td>\n",
       "      <td>-0.192420</td>\n",
       "      <td>-0.445430</td>\n",
       "      <td>-0.102640</td>\n",
       "      <td>0.19628</td>\n",
       "      <td>-0.095456</td>\n",
       "      <td>-1.399500</td>\n",
       "      <td>-0.478150</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.100140</td>\n",
       "      <td>-0.676610</td>\n",
       "      <td>0.304890</td>\n",
       "      <td>-0.540600</td>\n",
       "      <td>-0.334280</td>\n",
       "      <td>1.229900</td>\n",
       "      <td>-0.328632</td>\n",
       "      <td>-0.330245</td>\n",
       "      <td>-1.254223</td>\n",
       "      <td>0.192863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1792</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.880500</td>\n",
       "      <td>0.604240</td>\n",
       "      <td>-0.310380</td>\n",
       "      <td>0.601800</td>\n",
       "      <td>-0.180450</td>\n",
       "      <td>0.11812</td>\n",
       "      <td>-0.202220</td>\n",
       "      <td>-2.182900</td>\n",
       "      <td>1.554200</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.586610</td>\n",
       "      <td>0.469800</td>\n",
       "      <td>-0.166270</td>\n",
       "      <td>-0.920470</td>\n",
       "      <td>-0.943470</td>\n",
       "      <td>0.278090</td>\n",
       "      <td>-0.948951</td>\n",
       "      <td>-0.073959</td>\n",
       "      <td>-0.796032</td>\n",
       "      <td>-0.208793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6377</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.212780</td>\n",
       "      <td>0.574020</td>\n",
       "      <td>-0.323870</td>\n",
       "      <td>-0.513780</td>\n",
       "      <td>-0.263550</td>\n",
       "      <td>0.92589</td>\n",
       "      <td>-0.153070</td>\n",
       "      <td>0.321780</td>\n",
       "      <td>0.091441</td>\n",
       "      <td>...</td>\n",
       "      <td>0.297383</td>\n",
       "      <td>-0.486600</td>\n",
       "      <td>-0.330460</td>\n",
       "      <td>-1.106000</td>\n",
       "      <td>-0.150950</td>\n",
       "      <td>0.589190</td>\n",
       "      <td>-1.291660</td>\n",
       "      <td>-0.892555</td>\n",
       "      <td>-0.931502</td>\n",
       "      <td>0.991503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004928</td>\n",
       "      <td>0.273890</td>\n",
       "      <td>-0.357320</td>\n",
       "      <td>-0.472230</td>\n",
       "      <td>0.188780</td>\n",
       "      <td>1.94850</td>\n",
       "      <td>-0.110500</td>\n",
       "      <td>0.136650</td>\n",
       "      <td>-0.000208</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.060720</td>\n",
       "      <td>-0.334700</td>\n",
       "      <td>-0.248240</td>\n",
       "      <td>-1.378500</td>\n",
       "      <td>-0.026322</td>\n",
       "      <td>0.213700</td>\n",
       "      <td>-0.524764</td>\n",
       "      <td>-0.613327</td>\n",
       "      <td>-1.288974</td>\n",
       "      <td>0.253337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5750</th>\n",
       "      <td>27.0</td>\n",
       "      <td>0.002211</td>\n",
       "      <td>0.074397</td>\n",
       "      <td>0.670265</td>\n",
       "      <td>-0.121499</td>\n",
       "      <td>-0.282985</td>\n",
       "      <td>-1.47840</td>\n",
       "      <td>-0.092727</td>\n",
       "      <td>-0.218190</td>\n",
       "      <td>-0.157340</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.079172</td>\n",
       "      <td>-0.158024</td>\n",
       "      <td>-0.058338</td>\n",
       "      <td>-0.231340</td>\n",
       "      <td>0.260340</td>\n",
       "      <td>-0.322565</td>\n",
       "      <td>0.869280</td>\n",
       "      <td>0.122838</td>\n",
       "      <td>-0.288481</td>\n",
       "      <td>-0.661225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2624</th>\n",
       "      <td>27.0</td>\n",
       "      <td>0.071701</td>\n",
       "      <td>0.021365</td>\n",
       "      <td>-0.206400</td>\n",
       "      <td>-0.440120</td>\n",
       "      <td>-0.247860</td>\n",
       "      <td>0.21098</td>\n",
       "      <td>-0.434760</td>\n",
       "      <td>-0.258630</td>\n",
       "      <td>-0.064565</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.307446</td>\n",
       "      <td>-0.177410</td>\n",
       "      <td>0.123340</td>\n",
       "      <td>0.190050</td>\n",
       "      <td>-0.344350</td>\n",
       "      <td>-0.332820</td>\n",
       "      <td>-0.788958</td>\n",
       "      <td>-0.064927</td>\n",
       "      <td>-1.246426</td>\n",
       "      <td>-0.502380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5007</th>\n",
       "      <td>27.0</td>\n",
       "      <td>0.006940</td>\n",
       "      <td>-0.064350</td>\n",
       "      <td>-0.074589</td>\n",
       "      <td>0.198150</td>\n",
       "      <td>-0.141280</td>\n",
       "      <td>0.30652</td>\n",
       "      <td>-0.319210</td>\n",
       "      <td>-0.613640</td>\n",
       "      <td>-0.160340</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.108959</td>\n",
       "      <td>-0.065061</td>\n",
       "      <td>0.339020</td>\n",
       "      <td>0.419500</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>-0.157810</td>\n",
       "      <td>-0.014350</td>\n",
       "      <td>-0.172026</td>\n",
       "      <td>-1.309777</td>\n",
       "      <td>-0.239612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6925</th>\n",
       "      <td>27.0</td>\n",
       "      <td>0.131440</td>\n",
       "      <td>-0.078326</td>\n",
       "      <td>0.308450</td>\n",
       "      <td>0.464070</td>\n",
       "      <td>-0.370510</td>\n",
       "      <td>-1.33630</td>\n",
       "      <td>-0.191930</td>\n",
       "      <td>-1.317700</td>\n",
       "      <td>-0.028137</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.072274</td>\n",
       "      <td>0.038092</td>\n",
       "      <td>0.406310</td>\n",
       "      <td>-0.303230</td>\n",
       "      <td>0.127870</td>\n",
       "      <td>-0.026296</td>\n",
       "      <td>-0.806935</td>\n",
       "      <td>-0.084823</td>\n",
       "      <td>-1.404030</td>\n",
       "      <td>0.284475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6558</th>\n",
       "      <td>27.0</td>\n",
       "      <td>0.022872</td>\n",
       "      <td>-0.056505</td>\n",
       "      <td>0.051972</td>\n",
       "      <td>0.425150</td>\n",
       "      <td>-0.552320</td>\n",
       "      <td>-0.38526</td>\n",
       "      <td>-0.095467</td>\n",
       "      <td>-0.552380</td>\n",
       "      <td>0.720380</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.115636</td>\n",
       "      <td>0.169140</td>\n",
       "      <td>0.088572</td>\n",
       "      <td>-0.238660</td>\n",
       "      <td>-0.333030</td>\n",
       "      <td>-0.057865</td>\n",
       "      <td>-0.065283</td>\n",
       "      <td>0.116597</td>\n",
       "      <td>-0.845437</td>\n",
       "      <td>-0.394365</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7783 rows × 457 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      project_id  1433BETA  1433EPSILON  1433ZETA     4EBP1  4EBP1_pS65  \\\n",
       "5424         0.0  0.066729     0.412330 -0.229260 -0.034305   -0.030930   \n",
       "5207         0.0  0.218150    -0.096406 -0.192420 -0.445430   -0.102640   \n",
       "1792         0.0  1.880500     0.604240 -0.310380  0.601800   -0.180450   \n",
       "6377         0.0  0.212780     0.574020 -0.323870 -0.513780   -0.263550   \n",
       "259          0.0  0.004928     0.273890 -0.357320 -0.472230    0.188780   \n",
       "...          ...       ...          ...       ...       ...         ...   \n",
       "5750        27.0  0.002211     0.074397  0.670265 -0.121499   -0.282985   \n",
       "2624        27.0  0.071701     0.021365 -0.206400 -0.440120   -0.247860   \n",
       "5007        27.0  0.006940    -0.064350 -0.074589  0.198150   -0.141280   \n",
       "6925        27.0  0.131440    -0.078326  0.308450  0.464070   -0.370510   \n",
       "6558        27.0  0.022872    -0.056505  0.051972  0.425150   -0.552320   \n",
       "\n",
       "      4EBP1_pT37T46  4EBP1_pT70     53BP1  ACC_pS79  ...       XPF     XRCC1  \\\n",
       "5424        0.71239   -0.240950 -0.916615 -0.285390  ...  0.436293 -0.361380   \n",
       "5207        0.19628   -0.095456 -1.399500 -0.478150  ... -0.100140 -0.676610   \n",
       "1792        0.11812   -0.202220 -2.182900  1.554200  ... -0.586610  0.469800   \n",
       "6377        0.92589   -0.153070  0.321780  0.091441  ...  0.297383 -0.486600   \n",
       "259         1.94850   -0.110500  0.136650 -0.000208  ... -0.060720 -0.334700   \n",
       "...             ...         ...       ...       ...  ...       ...       ...   \n",
       "5750       -1.47840   -0.092727 -0.218190 -0.157340  ... -0.079172 -0.158024   \n",
       "2624        0.21098   -0.434760 -0.258630 -0.064565  ... -0.307446 -0.177410   \n",
       "5007        0.30652   -0.319210 -0.613640 -0.160340  ... -0.108959 -0.065061   \n",
       "6925       -1.33630   -0.191930 -1.317700 -0.028137  ... -0.072274  0.038092   \n",
       "6558       -0.38526   -0.095467 -0.552380  0.720380  ... -0.115636  0.169140   \n",
       "\n",
       "           YAP  YAP_pS127       YB1  YB1_pS102    YTHDF2    YTHDF3    ZAP.70  \\\n",
       "5424  0.106262  -0.529325 -0.530005   0.002020 -0.397034 -0.382803 -1.629115   \n",
       "5207  0.304890  -0.540600 -0.334280   1.229900 -0.328632 -0.330245 -1.254223   \n",
       "1792 -0.166270  -0.920470 -0.943470   0.278090 -0.948951 -0.073959 -0.796032   \n",
       "6377 -0.330460  -1.106000 -0.150950   0.589190 -1.291660 -0.892555 -0.931502   \n",
       "259  -0.248240  -1.378500 -0.026322   0.213700 -0.524764 -0.613327 -1.288974   \n",
       "...        ...        ...       ...        ...       ...       ...       ...   \n",
       "5750 -0.058338  -0.231340  0.260340  -0.322565  0.869280  0.122838 -0.288481   \n",
       "2624  0.123340   0.190050 -0.344350  -0.332820 -0.788958 -0.064927 -1.246426   \n",
       "5007  0.339020   0.419500  0.000640  -0.157810 -0.014350 -0.172026 -1.309777   \n",
       "6925  0.406310  -0.303230  0.127870  -0.026296 -0.806935 -0.084823 -1.404030   \n",
       "6558  0.088572  -0.238660 -0.333030  -0.057865 -0.065283  0.116597 -0.845437   \n",
       "\n",
       "          ZEB1  \n",
       "5424  1.311241  \n",
       "5207  0.192863  \n",
       "1792 -0.208793  \n",
       "6377  0.991503  \n",
       "259   0.253337  \n",
       "...        ...  \n",
       "5750 -0.661225  \n",
       "2624 -0.502380  \n",
       "5007 -0.239612  \n",
       "6925  0.284475  \n",
       "6558 -0.394365  \n",
       "\n",
       "[7783 rows x 457 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Reading in full TCGA data which contains all the proteins\n",
    "Full_TCGA_data_ori = pd.read_csv(\"../R/Data/Processed_Data/KNN_filled_data.csv\")\n",
    "Full_TCGA_data_ori.drop(['Unnamed: 0'], axis=1, inplace=True)\n",
    "\n",
    "Full_TCGA_data_ori = Full_TCGA_data_ori.sort_values(by='project_id')\n",
    "\n",
    "\n",
    "Full_TCGA_data_ori"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating dataset with the shortlisted proteins from the paper"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**First do this before the following code:** From the reference paper, download Supplementary Data 9 and put it into the folder R/Data/Processed_Data.\n",
    "\n",
    "**NOTE:** ensure that the 'Pan-can ab list' sheet is the first sheet in the excel just downloaded."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Eloisa\\AppData\\Roaming\\Python\\Python312\\site-packages\\openpyxl\\worksheet\\_reader.py:329: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#</th>\n",
       "      <th>Protein Name</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Antibody validation status</th>\n",
       "      <th>Antibody Origin</th>\n",
       "      <th>Antibody Source (Company)</th>\n",
       "      <th>Catalog Number</th>\n",
       "      <th>Dilution</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>4E-BP1_pS65</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>CST</td>\n",
       "      <td>9456</td>\n",
       "      <td>1:250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>4E-BP1_pT37</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>CST</td>\n",
       "      <td>9459</td>\n",
       "      <td>1:100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4E-BP1</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>CST</td>\n",
       "      <td>9452</td>\n",
       "      <td>1:100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>ACC_pS79</td>\n",
       "      <td>ACACA ACACB</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>CST</td>\n",
       "      <td>3661</td>\n",
       "      <td>1:250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>Akt_pS473</td>\n",
       "      <td>AKT1 AKT2 AKT3</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>CST</td>\n",
       "      <td>9271</td>\n",
       "      <td>1:250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>171</td>\n",
       "      <td>Transglutaminase</td>\n",
       "      <td>TGM2</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Mouse</td>\n",
       "      <td>Lab Vision</td>\n",
       "      <td>MS-224</td>\n",
       "      <td>1:100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>172</td>\n",
       "      <td>TFRC</td>\n",
       "      <td>TFRC</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>SDI / Novus</td>\n",
       "      <td>22500002</td>\n",
       "      <td>1:100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>174</td>\n",
       "      <td>Tuberin_pT1462</td>\n",
       "      <td>TSC2</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>CST</td>\n",
       "      <td>3617</td>\n",
       "      <td>1:500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>180</td>\n",
       "      <td>ETS-1</td>\n",
       "      <td>ETS-1</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>BethYl</td>\n",
       "      <td>A303-501A</td>\n",
       "      <td>1:750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>181</td>\n",
       "      <td>Rab25</td>\n",
       "      <td>Rab 25</td>\n",
       "      <td>Validated</td>\n",
       "      <td>Rabbit</td>\n",
       "      <td>CST</td>\n",
       "      <td>4314</td>\n",
       "      <td>1:250</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>113 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0      #      Protein Name       Gene Name Antibody validation status  \\\n",
       "2      2       4E-BP1_pS65        EIF4EBP1                  Validated   \n",
       "3      3       4E-BP1_pT37        EIF4EBP1                  Validated   \n",
       "4      4            4E-BP1        EIF4EBP1                  Validated   \n",
       "6      6          ACC_pS79     ACACA ACACB                  Validated   \n",
       "8      8         Akt_pS473  AKT1 AKT2 AKT3                  Validated   \n",
       "..   ...               ...             ...                        ...   \n",
       "171  171  Transglutaminase            TGM2                  Validated   \n",
       "172  172              TFRC            TFRC                  Validated   \n",
       "174  174    Tuberin_pT1462            TSC2                  Validated   \n",
       "180  180             ETS-1           ETS-1                  Validated   \n",
       "181  181             Rab25          Rab 25                  Validated   \n",
       "\n",
       "0   Antibody Origin Antibody Source (Company) Catalog Number Dilution  \n",
       "2            Rabbit                       CST           9456    1:250  \n",
       "3            Rabbit                       CST           9459    1:100  \n",
       "4            Rabbit                       CST           9452    1:100  \n",
       "6            Rabbit                       CST           3661    1:250  \n",
       "8            Rabbit                       CST           9271    1:250  \n",
       "..              ...                       ...            ...      ...  \n",
       "171           Mouse                Lab Vision         MS-224    1:100  \n",
       "172          Rabbit               SDI / Novus       22500002    1:100  \n",
       "174          Rabbit                       CST           3617    1:500  \n",
       "180          Rabbit                    BethYl      A303-501A    1:750  \n",
       "181          Rabbit                       CST           4314    1:250  \n",
       "\n",
       "[113 rows x 8 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# reads by default 1st sheet of an excel file\n",
    "short_protein_list = pd.read_excel(\"../R/Data/Processed_Data/41467_2014_BFncomms4887_MOESM464_ESM.xlsx\")\n",
    "\n",
    "# Set a specific row as column names\n",
    "short_protein_list.columns = short_protein_list.iloc[0]\n",
    "\n",
    "# Drop the row that was used as column names\n",
    "short_protein_list = short_protein_list[1:]\n",
    "\n",
    "# Keeping only validated proteins\n",
    "short_protein_list = short_protein_list[short_protein_list['Antibody validation status'] == 'Validated']\n",
    "\n",
    "short_protein_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False occurs 4 times\n",
      "True occurs 109 times\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Eloisa\\AppData\\Roaming\\Python\\Python312\\site-packages\\openpyxl\\worksheet\\_reader.py:329: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    }
   ],
   "source": [
    "# reads by default 1st sheet of an excel file\n",
    "short_protein_list = pd.read_excel(\"../R/Data/Processed_Data/41467_2014_BFncomms4887_MOESM464_ESM.xlsx\")\n",
    "\n",
    "# Setting a specific row as column names\n",
    "short_protein_list.columns = short_protein_list.iloc[0]\n",
    "\n",
    "# Drop the row that was used as column names\n",
    "short_protein_list = short_protein_list[1:]\n",
    "\n",
    "# Keeping only validated proteins\n",
    "short_protein_list = short_protein_list[short_protein_list['Antibody validation status'] == 'Validated']\n",
    "\n",
    "#short_protein_list1 = short_protein_list['Official Ab Name ']\n",
    "short_protein_list = short_protein_list['Protein Name']\n",
    "short_protein_list = [x for x in short_protein_list if str(x) != 'nan']\n",
    "\n",
    "\n",
    "# ----------------------------------------------------------------------------\n",
    "# Ensuring that the protein names in my data frame and their list match\n",
    "short_protein_list = pd.DataFrame(short_protein_list)\n",
    "short_protein_list.replace({'-':''}, regex=True, inplace=True)\n",
    "short_protein_list.replace({'_':''}, regex=True, inplace=True)\n",
    "short_protein_list.replace({' ':''}, regex=True, inplace=True)\n",
    "short_protein_list = short_protein_list[0].str.lower()\n",
    "short_protein_list = short_protein_list.str.replace(r'\\W', '')\n",
    "\n",
    "# Making new full dataset variable to keep orignial which has no changes to the column names\n",
    "Full_TCGA_data = Full_TCGA_data_ori\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(r'\\W', '')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('-','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('_','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(' ','')\n",
    "\n",
    "\n",
    "# Checking how many names in the paper's protein list is in my data frame\n",
    "common_proteins1 = short_protein_list.isin(Full_TCGA_data.columns.str.lower())\n",
    "unique_values, counts = np.unique(common_proteins1, return_counts=True)\n",
    "\n",
    "for value, count in zip(unique_values, counts):\n",
    "    print(f\"{value} occurs {count} times\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Since 5 proteins were not found in common, I will check if the protein name they used is called something different by looking at the second sheet called 'Standard Ab List_Website' which contains an 'Official Ab name' column corresponding to the protein list they used to name their proteins. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Exporting the uncommon proteins for manual edit\n",
    "common_proteins1 = pd.DataFrame(common_proteins1)\n",
    "X = pd.DataFrame(common_proteins1[common_proteins1[0]== False])\n",
    "index = X.index.values\n",
    "y1 = short_protein_list.iloc[index]\n",
    "y1.to_csv(\"../R/Data/Processed_Data/Not_common_proteins.csv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True occurs 4 times\n"
     ]
    }
   ],
   "source": [
    "# Importing the manually edited version\n",
    "found_proteins1 = pd.read_csv(\"../R/Data/Processed_Data/Not_common_proteins_manual_edit.csv\")\n",
    "found_proteins = pd.DataFrame(found_proteins1['protein'])\n",
    "\n",
    "found_proteins.replace({'-':''}, regex=True, inplace=True)\n",
    "found_proteins.replace({'_':''}, regex=True, inplace=True)\n",
    "found_proteins.replace({' ':''}, regex=True, inplace=True)\n",
    "found_proteins = found_proteins['protein'].str.lower()\n",
    "found_proteins = found_proteins.str.replace(r'\\W', '')\n",
    "\n",
    "common_proteins2 = found_proteins.isin(Full_TCGA_data.columns.str.lower())\n",
    "\n",
    "unique_values, counts = np.unique(common_proteins2, return_counts=True)\n",
    "\n",
    "\n",
    "# Print the results\n",
    "for value, count in zip(unique_values, counts):\n",
    "    print(f\"{value} occurs {count} times\")\n",
    "\n",
    "# From doing the mannual edit I was able to identify them in my dataset"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All 113 validated proteins were found in my dataset."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now I can pull out these proteins from the full TCGA dataset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True occurs 113 times\n"
     ]
    }
   ],
   "source": [
    "X1 = pd.DataFrame(common_proteins1[common_proteins1[0]== True])\n",
    "index = X1.index.values\n",
    "y1 = short_protein_list.iloc[index]\n",
    "\n",
    "common_proteins2 = pd.DataFrame(common_proteins2)\n",
    "X2 = pd.DataFrame(common_proteins2[common_proteins2['protein']== True])\n",
    "index = X2.index.values\n",
    "y2 = found_proteins1['protein'].iloc[index]\n",
    "\n",
    "\n",
    "y = pd.concat([y1,y2], ignore_index=True)\n",
    "\n",
    "y.replace({'-':''}, regex=True, inplace=True)\n",
    "y.replace({'_':''}, regex=True, inplace=True)\n",
    "y.replace({' ':''}, regex=True, inplace=True)\n",
    "y = y.str.lower()\n",
    "y = y.str.replace(r'\\W', '')\n",
    "\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(r'\\W', '')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('-','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('_','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(' ','')\n",
    "\n",
    "common = y.isin(Full_TCGA_data.columns.str.lower())\n",
    "\n",
    "unique_values, counts = np.unique(common, return_counts=True)\n",
    "\n",
    "# Print the results\n",
    "for value, count in zip(unique_values, counts):\n",
    "    print(f\"{value} occurs {count} times\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Producing the shortlisted Full TCGA dataframe and saving this\n",
    "y = list(y)\n",
    "y.insert(0, Full_TCGA_data_ori.columns[0])\n",
    "y\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(r'\\W', '')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('-','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('_','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(' ','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.lower() \n",
    "\n",
    "\n",
    "shortlisted_TCGA_data = Full_TCGA_data[y]\n",
    "shortlisted_TCGA_data.to_csv(\"../R/Data/Processed_Data/Shortlisted_protein_list.csv\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating a machine learning and validation TCGA datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "\n",
    "# Group the DataFrame by 'CancerType'\n",
    "grouped_df = shortlisted_TCGA_data.groupby('projectid')\n",
    "\n",
    "# Create empty DataFrames for the 70% and 30% splits\n",
    "ML_70_percent = pd.DataFrame()\n",
    "V_30_percent = pd.DataFrame()\n",
    "\n",
    "# Iterate over groups and split each group\n",
    "for group_name, group_data in grouped_df:\n",
    "    group_data_70, group_data_30 = train_test_split(group_data, test_size=0.3, random_state=42)\n",
    "    ML_70_percent = pd.concat([group_data_70, ML_70_percent])\n",
    "    V_30_percent = pd.concat([group_data_30, V_30_percent])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Saving the final machine learning and validation dataset\n",
    "ML_70_percent = ML_70_percent.sort_values(by='projectid')\n",
    "ML_70_percent.to_csv(\"../R/Data/Processed_Data/Final_Machine_Learning_Data.csv\")\n",
    "V_30_percent = V_30_percent.sort_values(by='projectid')\n",
    "V_30_percent.to_csv(\"../R/Data/Processed_Data/Final_Validation_Data.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating a dataset which contains the validated proteins and their associated gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Eloisa\\AppData\\Roaming\\Python\\Python312\\site-packages\\openpyxl\\worksheet\\_reader.py:329: UserWarning: Unknown extension is not supported and will be removed\n",
      "  warn(msg)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein Name</th>\n",
       "      <th>Gene Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4ebp1ps65</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4ebp1pt37</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4ebp1</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>accps79</td>\n",
       "      <td>ACACA ACACB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>aktps473</td>\n",
       "      <td>AKT1 AKT2 AKT3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>transglutaminase</td>\n",
       "      <td>TGM2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>tfrc</td>\n",
       "      <td>TFRC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>tuberinpt1462</td>\n",
       "      <td>TSC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>180</th>\n",
       "      <td>ets1</td>\n",
       "      <td>ETS-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>181</th>\n",
       "      <td>rab25</td>\n",
       "      <td>Rab 25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>113 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "0        Protein Name       Gene Name\n",
       "2           4ebp1ps65        EIF4EBP1\n",
       "3           4ebp1pt37        EIF4EBP1\n",
       "4               4ebp1        EIF4EBP1\n",
       "6             accps79     ACACA ACACB\n",
       "8            aktps473  AKT1 AKT2 AKT3\n",
       "..                ...             ...\n",
       "171  transglutaminase            TGM2\n",
       "172              tfrc            TFRC\n",
       "174     tuberinpt1462            TSC2\n",
       "180              ets1           ETS-1\n",
       "181             rab25          Rab 25\n",
       "\n",
       "[113 rows x 2 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_gene_list =  pd.read_excel(\"../R/Data/Processed_Data/41467_2014_BFncomms4887_MOESM464_ESM.xlsx\")\n",
    "\n",
    "# Setting a specific row as column names\n",
    "protein_gene_list.columns = protein_gene_list.iloc[0]\n",
    "\n",
    "# Drop the row that was used as column names\n",
    "protein_gene_list = protein_gene_list[1:]\n",
    "\n",
    "# Keeping only validated proteins\n",
    "protein_gene_list = protein_gene_list[protein_gene_list['Antibody validation status'] == 'Validated']\n",
    "\n",
    "protein_gene_list = protein_gene_list[['Protein Name', 'Gene Name']]\n",
    "\n",
    "\n",
    "protein_gene_list\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True occurs 113 times\n"
     ]
    }
   ],
   "source": [
    "# Ensuring names are in the same format\n",
    "protein_gene_list['Protein Name'] = protein_gene_list['Protein Name'].str.replace(r'\\W', '')\n",
    "protein_gene_list['Protein Name'] = protein_gene_list['Protein Name'].str.replace('-','')\n",
    "protein_gene_list['Protein Name'] = protein_gene_list['Protein Name'].str.replace('_','')\n",
    "protein_gene_list['Protein Name'] = protein_gene_list['Protein Name'].str.replace(' ','')\n",
    "protein_gene_list['Protein Name'] = protein_gene_list['Protein Name'].str.lower() \n",
    "\n",
    "# Create a dictionary mapping the different protein names to those found in my dataset\n",
    "replacement_dict = dict(zip(found_proteins1['0'], found_proteins1['protein']))\n",
    "\n",
    "# Replace names based on the mapping in replacement_dict\n",
    "protein_gene_list['Protein Name'] = protein_gene_list['Protein Name'].replace(replacement_dict)\n",
    "\n",
    "#-------------------------------------------------------------------------------------------------#\n",
    "# Double mapping was done correctly\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(r'\\W', '')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('-','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace('_','')\n",
    "Full_TCGA_data.columns = Full_TCGA_data.columns.str.replace(' ','')\n",
    "\n",
    "common1 = protein_gene_list['Protein Name'].isin(Full_TCGA_data.columns.str.lower())\n",
    "\n",
    "unique_values, counts = np.unique(common1, return_counts=True)\n",
    "\n",
    "# Print the results\n",
    "for value, count in zip(unique_values, counts):\n",
    "    print(f\"{value} occurs {count} times\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now the protein to gene dataframe contains protein names found in my dataset. This is required as this dataset will be used as reference to create the protein interactions matrix. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating a unique gene list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creating a protein list which contains unique gene names \n",
    "unique_list = []\n",
    "\n",
    "for value in protein_gene_list['Gene Name']:\n",
    "    unique_list.extend(value.split())\n",
    "\n",
    "unique_list = list(set(unique_list))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_list = pd.DataFrame(unique_list)\n",
    "unique_list.to_csv(\"../R/Data/Processed_Data/Shortlisted_unique_gene_list.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creating a protein-gene conversion list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is necessary as in the R script it needs a protein-gene conversion dataframe which has each individual gene in its own box in the dataframe."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein Name</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4ebp1ps65</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4ebp1pt37t46</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4ebp1</td>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>accps79</td>\n",
       "      <td>ACACA</td>\n",
       "      <td>ACACB</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>aktps473</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>AKT2</td>\n",
       "      <td>AKT3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Protein Name         0      1     2\n",
       "2     4ebp1ps65  EIF4EBP1   None  None\n",
       "3  4ebp1pt37t46  EIF4EBP1   None  None\n",
       "4         4ebp1  EIF4EBP1   None  None\n",
       "6       accps79     ACACA  ACACB  None\n",
       "8      aktps473      AKT1   AKT2  AKT3"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene = protein_gene_list['Gene Name'].str.split(expand=True)\n",
    "protein_to_gene = pd.concat([protein_gene_list['Protein Name'], gene], axis=1)\n",
    "protein_to_gene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_to_gene.to_csv(\"../R/Data/Processed_Data/Protein_to_gene_conversion.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
